File: backend/app/schemas/notification.py
from datetime import datetime

from pydantic import BaseModel


class NotificationResponse(BaseModel):
    id: str
    user_id: str | None
    role: str | None
    title: str
    message: str
    event_type: str
    related_entity_type: str | None
    related_entity_id: str | None
    is_read: bool
    created_at: datetime
    read_at: datetime | None

File: frontend/src/types/notification.ts
export type Notification = {
  id: string;
  user_id: string | null;
  role: string | null;
  title: string;
  message: string;
  event_type: string;
  related_entity_type: string | null;
  related_entity_id: string | null;
  is_read: boolean;
  created_at: string;
  read_at: string | null;
};

